src/recv_udp_pcap.cpp

```diff
@@ -108,8 +108,11 @@ udp_pcap_file_reader::udp_pcap_file_reader(stream &owner, const std::string &fil
         throw std::runtime_error(errbuf);
     // Set a filter to ensure that we only get UDP4 packets with no fragmentation
     bpf_program filter;
+    std::string filter_expression = "ip proto \\udp and ip[6:2] & 0x3fff = 0";
+    if (!user_filter.empty())
+        filter_expression += " and (" + user_filter + ')';
     if (pcap_compile(handle, &filter,
-                     "ip proto \\udp and ip[6:2] & 0x3fff = 0",
+                     filter_expression.data(),
```

Comment:
```suggestion
                     filter_expression.c_str(),
```
I know they're the same since C++11, but subconsciously I still look at `.data()` and think of it as not being null-terminated.